Descent3/Mission.cpp

```diff
@@ -742,6 +745,32 @@ void InitMission() {
   }
   atexit(ResetMission);
 }
+
+void InitDefaultMissionFromCLI() {
+  // Load mission on startup
+  int mission_arg = FindArg("-mission");
+  if (mission_arg > 0) {
+    char path_to_mission[_MAX_PATH];
+    char filename[256];
+
+    // get the true filename
+    ddio_SplitPath(GameArgs[mission_arg + 1], NULL, filename, NULL);
+    strcat(filename, ".mn3");
+
+    LoadMission(filename);
+
+    int level_arg = FindArg("-loadlevel");
+    if (level_arg > 0) {
+      const int level{std::atoi(GameArgs[level_arg + 1])};
+      if (level <= 0 || level > Current_mission.num_levels) {
+        Error("Could not load level %d", level);
+      } else {
+        Current_mission.cur_level = level;
+      }
```

Comment:
Probably instead of error better to setting `level = 1` and throw warning into log:

```
if (level <= 0 || level > Current_mission.num_levels) {
  LOG.WARNING("Invalid level selection, resetting to 1");
  level = 1;
}
Current_mission.cur_level = level;
```